src/legacy/index.ts

```diff
@@ -214,9 +235,13 @@ function constructTargetOS(depGraph: types.DepGraph): { name: string; version: s
 async function buildSubtree(
     depGraph: types.DepGraphInternal, nodeId: string, eventLoopSpinner: EventLoopSpinner): Promise<DepTree> {
   const nodePkg = depGraph.getNodePkg(nodeId);
+  const nodeInfo = depGraph.getNode(nodeId);
   const depTree: DepTree = {};
   depTree.name = nodePkg.name;
   depTree.version = nodePkg.version;
+  if (nodeInfo.versionProvenance) {
+    depTree.versionProvenance = nodeInfo.versionProvenance;
```

Comment:
should we be worried about hanging on to a reference here? potential source of bugs...